js/ui/components/clubhouse.js

```diff
@@ -148,6 +153,111 @@ var ClubhouseWindowTracker = new Lang.Class({
     },
 });
 
+var getClubhouseAnimator = (function () {
+    let clubhouseAnimator;
+    return function() {
+        if (!clubhouseAnimator)
+            clubhouseAnimator = new ClubhouseAnimator();
+        return clubhouseAnimator;
+    };
+}());
+
+var ClubhouseAnimator = new Lang.Class({
+    Name: 'ClubhouseAnimator',
+    Extends: GObject.Object,
+
+    Signals: {
+        'animation-ready': {},
+    },
+
+    _init: function() {
+        this.parent();
+        this._proxy = null;
+        this._animations = {};
+        this._clubhousePaths = this._getClubhousePaths();
+    },
+
+    setClubhouseProxy: function(proxy) {
+        this._proxy = proxy;
+    },
+
+    _getClubhousePaths: function() {
+        let paths = [];
+        let installations = Flatpak.get_system_installations(null);
+        installations.unshift(Flatpak.Installation.new_user(null));
+
+        for (let installation of installations) {
+            let app = null;
+            try {
+                app = installation.get_current_installed_app(CLUBHOUSE_ID, null);
+            } catch (err if err.matches(Flatpak.Error, Flatpak.Error.NOT_INSTALLED)) {
+                continue;
+            }
+
+            if (app) {
+                let deploy_dir = app.get_deploy_dir();
+                paths.push(this._getActivateDir(deploy_dir));
+            }
+        }
+
+        return paths;
+    },
+
+    _getActivateDir: function(deployDir) {
+        // Replace the hash part of the deploy directory by "active", so the directory
+        // is always the most recent one (i.e. allows us to update the Clubhouse and
+        // still have the right dir).
+        let dir = deployDir.substr(-1) == '/' ? deployDir.slice(0, -1) : deployDir;
+        let splitDir = dir.split('/')
+
+        splitDir[splitDir.length - 1] = 'active';
+
+        return splitDir.join('/');
+    },
+
+    _getClubhouseUri: function(uri) {
+        // Discard the /app/ prefix
+        let uriSuffix = uri.replace(/^\/app\//g, '');
+
+        for (let path of this._clubhousePaths) {
+            let completeUri = path + '/files/' + uriSuffix;
+            if (GLib.file_test(completeUri, GLib.FileTest.EXISTS))
+                return completeUri;
+        }
+
+        return null;
+    },
+
+    getAnimationByUri: function(uri) {
```

Comment:
I have just realized I should disconnect from the `animation-ready` signal when the banner is dismissed to avoid the cb eventually getting called if in the future we use animations in other places (the animator is a singleton).

I don't see how to add a cb to this function in this case, as it'd require more changes to the animator, but if I am seeing it more complicated than it is, please let me know.